tests/test_scalars.py

```diff
@@ -339,147 +381,219 @@ def test_timedelta() -> None:
+    np_float_arr: npt.NDArray[np.floating] = np.array([1, 2, 3])
     check(assert_type(td // td, int), int)
     check(assert_type(td // 3, pd.Timedelta), pd.Timedelta)
     check(assert_type(td // 3.5, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td // np.array([1, 2, 3]), np.ndarray), np.ndarray)
-    check(assert_type(td // np.array([1.2, 2.2, 3.4]), np.ndarray), np.ndarray)
+    check(assert_type(td // np_intp_arr, npt.NDArray[np.timedelta64]), np.ndarray)
+    check(assert_type(td // np_float_arr, npt.NDArray[np.timedelta64]), np.ndarray)
     check(assert_type(td // pd.Series([1, 2, 3]), pd.Series), pd.Series)
     check(assert_type(td // pd.Series([1.2, 2.2, 3.4]), pd.Series), pd.Series)
-    check(assert_type(td // pd.Index([1, 2, 3]), pd.TimedeltaIndex), pd.TimedeltaIndex)
-    check(
-        assert_type(td // pd.Index([1.2, 2.2, 3.4]), pd.TimedeltaIndex),
-        pd.TimedeltaIndex,
-    )
+    check(assert_type(td // i_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td // f_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+
+    # Note: None of the reverse floordiv work
+    # check(assert_type(3 // td, pd.Timedelta), pd.Timedelta)
+    # check(assert_type(3.5// td, pd.Timedelta), pd.Timedelta)
+    # check(assert_type(np_intp_arr// td, npt.NDArray[np.timedelta64]), np.ndarray)
+    # check(assert_type(np_float_arr// td, npt.NDArray[np.timedelta64]), np.ndarray)
+    # check(assert_type(pd.Series([1, 2, 3])// td, pd.Series), pd.Series)
+    # check(assert_type(pd.Series([1.2, 2.2, 3.4])// td, pd.Series), pd.Series)
+    # check(assert_type(i_idx, pd.TimedeltaIndex)// td, pd.TimedeltaIndex)
+    # check( assert_type(f_idx// td, pd.TimedeltaIndex), pd.TimedeltaIndex)
 
     check(assert_type(td / td, float), float)
     check(assert_type(td / 3, pd.Timedelta), pd.Timedelta)
     check(assert_type(td / 3.5, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td / np.array([1, 2, 3]), np.ndarray), np.ndarray)
-    check(assert_type(td / np.array([1.2, 2.2, 3.4]), np.ndarray), np.ndarray)
-    check(assert_type(td / pd.Series([1, 2, 3]), pd.Series), pd.Series)
-    check(assert_type(td / pd.Series([1.2, 2.2, 3.4]), pd.Series), pd.Series)
-    check(assert_type(td / pd.Index([1, 2, 3]), pd.TimedeltaIndex), pd.TimedeltaIndex)
     check(
-        assert_type(td / pd.Index([1.2, 2.2, 3.4]), pd.TimedeltaIndex),
-        pd.TimedeltaIndex,
+        assert_type(td / np.array([1, 2, 3]), npt.NDArray[np.timedelta64]), np.ndarray
     )
+    check(
+        assert_type(td / np.array([1.2, 2.2, 3.4]), npt.NDArray[np.timedelta64]),
+        np.ndarray,
+    )
+    check(assert_type(td / pd.Series([1, 2, 3]), pd.Series), pd.Series)
+    check(assert_type(td / pd.Series([1.2, 2.2, 3.4]), pd.Series), pd.Series)
+    check(assert_type(td / i_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td / f_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+
+    # Note: None of the reverse truediv work
```

Comment:
same comment as above with respect to clarifying the comment and using `TYPE_CHECKING_INVALID_USAGE`